tests/src/test/scala/common/BaseWsk.scala

```diff
@@ -0,0 +1,365 @@
+    implicit wp: WskProps): RunResult
+}
+
+trait BaseNamespace extends BaseRunWsk {
+
+  def list(expectedExitCode: Int = SUCCESS_EXIT, nameSort: Option[Boolean] = None)(implicit wp: WskProps): RunResult
+
+  def whois()(implicit wskprops: WskProps): String
+
+  def get(namespace: Option[String] = None, expectedExitCode: Int, nameSort: Option[Boolean] = None)(
+    implicit wp: WskProps): RunResult
+}
+
+trait BaseApi extends BaseRunWsk {
+
+  def create(basepath: Option[String] = None,
+             relpath: Option[String] = None,
+             operation: Option[String] = None,
+             action: Option[String] = None,
+             apiname: Option[String] = None,
+             swagger: Option[String] = None,
+             responsetype: Option[String] = None,
+             expectedExitCode: Int = SUCCESS_EXIT,
+             cliCfgFile: Option[String] = None)(implicit wp: WskProps): RunResult
+
+  def list(basepathOrApiName: Option[String] = None,
+           relpath: Option[String] = None,
+           operation: Option[String] = None,
+           limit: Option[Int] = None,
+           since: Option[Instant] = None,
+           full: Option[Boolean] = None,
+           nameSort: Option[Boolean] = None,
+           expectedExitCode: Int = SUCCESS_EXIT,
+           cliCfgFile: Option[String] = None)(implicit wp: WskProps): RunResult
+
+  def get(basepathOrApiName: Option[String] = None,
+          full: Option[Boolean] = None,
+          expectedExitCode: Int = SUCCESS_EXIT,
+          cliCfgFile: Option[String] = None,
+          format: Option[String] = None)(implicit wp: WskProps): RunResult
+
+  def delete(basepathOrApiName: String,
+             relpath: Option[String] = None,
+             operation: Option[String] = None,
+             expectedExitCode: Int = SUCCESS_EXIT,
+             cliCfgFile: Option[String] = None)(implicit wp: WskProps): RunResult
+}
+
+trait BaseRunWsk extends Matchers {
+
+  // Takes a string and a list of sensitive strings. Any sensistive string found in
+  // the target string will be replaced with "XXXXX", returning the processed string
+  def hideStr(str: String, hideThese: Seq[String]): String = {
+    // Iterate through each string to hide, replacing it in the target string (str)
+    hideThese.fold(str)((updatedStr, replaceThis) => updatedStr.replace(replaceThis, "XXXXX"))
+  }
+
+  /*
+   * Utility function to return a JSON object from the CLI output that returns
+   * an optional a status line following by the JSON data
+   */
+  def parseJsonString(jsonStr: String): JsObject = {
+    jsonStr.substring(jsonStr.indexOf("\n") + 1).parseJson.asJsObject // Skip optional status line before parsing
+  }
+
+  def reportFailure(args: Buffer[String], ec: Integer, rr: RunResult) = {
+    val s = new StringBuilder()
+    s.append(args.mkString(" ") + "\n")
+    if (rr.stdout.nonEmpty) s.append(rr.stdout + "\n")
+    if (rr.stderr.nonEmpty) s.append(rr.stderr)
+    s.append("exit code:")
+  }
```

Comment:
Do all of these methods apply for both Rest and Cli? At least `parseJsonString` doesn't look like it.

Reply:
I would create a class named RestResult to save the REST results. In order to keep the changes in test cases as few as possible, I make RestResult extend RunResult, so WskTestHelpers can work for both CLI and REST without changes. RestResult can be converted into RunResult.